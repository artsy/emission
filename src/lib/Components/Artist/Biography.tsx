import * as React from "react"
import { Dimensions, StyleSheet, View, ViewProperties } from "react-native"
import { createFragmentContainer, graphql } from "react-relay"

import removeMarkdown from "remove-markdown"

import Headline from "../Text/Headline"
import SerifText from "../Text/Serif"

const sideMargin = Dimensions.get("window").width > 700 ? 50 : 0

interface Props extends ViewProperties {
  artist: any
}

class Biography extends React.Component<Props, any> {
  render() {
    const artist = this.props.artist
    if (!artist.blurb && !artist.bio) {
      return null
    }

    return (
      <View style={{ marginLeft: sideMargin, marginRight: sideMargin }}>
        <Headline style={{ marginBottom: 20 }}>Biography</Headline>
        {this.blurb(artist)}
        <SerifText style={styles.bio} numberOfLines={0}>
          {this.bioText()}
        </SerifText>
      </View>
    )
  }

  blurb(artist) {
    if (artist.blurb) {
      return (
        <SerifText style={styles.blurb} numberOfLines={0}>
          {removeMarkdown(artist.blurb)}
        </SerifText>
      )
    }
  }

  bioText() {
    const bio = this.props.artist.bio
    return bio.replace("born", "b.")
  }
}

const styles = StyleSheet.create({
  blurb: {
    fontSize: 16,
    lineHeight: 20,
    marginBottom: 20,
  },
  bio: {
    fontSize: 16,
    lineHeight: 20,
    marginBottom: 40,
  },
})

export default createFragmentContainer(
  Biography,
  graphql`
    fragment Biography_artist on Artist {
      bio
      blurb
    }
  `
)

interface RelayProps {
  artist: {
    bio: string | null
    blurb: string | null
  }
}
