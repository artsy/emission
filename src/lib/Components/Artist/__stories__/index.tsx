import "./ArtistArticles.story"
import "./ArtistHeader.story"
