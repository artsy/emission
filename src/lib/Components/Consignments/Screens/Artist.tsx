import * as React from "react"

import ArtistSearch from "../Components/ArtistSearchResults"
import DoneButton from "../Components/BottomAlignedButton"
import { ConsignmentSetup } from "../index"

import { NavigatorIOS, Route, ScrollView, View, ViewProperties } from "react-native"

interface Props extends ConsignmentSetup {
  navigator: NavigatorIOS
  route: Route
}

export default class Artist extends React.Component<Props, null> {
  doneTapped = () => {
    // Update state
    this.props.navigator.pop()
  }

  render() {
    return (
      <View style={{ backgroundColor: "black", flex: 1 }}>
        <DoneButton onPress={this.doneTapped}>
          <View
            style={{ alignContent: "center", justifyContent: "center", flexGrow: 1, marginLeft: 20, marginRight: 20 }}
          >
            <ArtistSearch results={null} query={null} />
          </View>
        </DoneButton>
      </View>
    )
  }
}
