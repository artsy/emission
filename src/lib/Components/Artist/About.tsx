import React from "react"
import { createFragmentContainer, graphql } from "react-relay"

import { Dimensions, StyleSheet, View } from "react-native"

import NavButton from "../Buttons/NavigationButton"
import RelatedArtists from "../RelatedArtists"
import Separator from "../Separator"
import Articles from "./Articles"
import Biography from "./Biography"

import { About_artist } from "__generated__/About_artist.graphql"

interface Props {
  artist: About_artist
}

class About extends React.Component<Props> {
  render() {
    return (
      <View>
        {this.biography()}
        {this.articles()}
        {this.relatedArtists()}
      </View>
    )
  }

  biography() {
    if (this.props.artist.has_metadata) {
      return (
        <View>
          <Biography artist={this.props.artist as any} />
          {this.auctionResults()}
          <Separator style={styles.sectionSeparator} />
        </View>
      )
    }
  }

  auctionResults() {
    if (this.props.artist.is_display_auction_link) {
      // Keeps the same margins as the bio text
      const sideMargin = Dimensions.get("window").width > 700 ? 50 : 20
      const url = `/artist/${this.props.artist.gravityID}/auction-results`
      return (
        <NavButton title="Auction Results" href={url} style={{ marginLeft: sideMargin, marginRight: sideMargin }} />
      )
    }
  }

  articles() {
    if (this.props.artist.articles.length) {
      return (
        <View>
          <Articles articles={this.props.artist.articles as any} />
          <Separator style={styles.sectionSeparator} />
        </View>
      )
    }
  }

  relatedArtists() {
    return this.props.artist.related_artists.length ? (
      <RelatedArtists artists={this.props.artist.related_artists as any} />
    ) : null
  }
}

const styles = StyleSheet.create({
  sectionSeparator: {
    marginBottom: 20,
  },
})

export default createFragmentContainer(About, {
  artist: graphql`
    fragment About_artist on Artist {
      has_metadata
      is_display_auction_link
      slug
      ...Biography_artist
      related_artists: artists(size: 16) {
        ...RelatedArtists_artists
      }
      articles {
        ...Articles_articles
      }
    }
  `,
})
