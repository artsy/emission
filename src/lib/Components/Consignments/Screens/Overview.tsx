import { Schema, screenTrack, Track, track as _track } from "lib/utils/track"
import React from "react"
import { AsyncStorage, Dimensions, NavigatorIOS, Route, ScrollView, View, ViewProperties } from "react-native"

import { ConsignmentMetadata, ConsignmentSetup, SearchResult } from "../"
import SwitchBoard from "../../../NativeModules/SwitchBoard"
import TODO from "../Components/ArtworkConsignmentTodo"
import CloseButton from "../Components/CloseButton"
import ConsignmentBG from "../Components/ConsignmentBG"
import { Button, Row } from "../Components/FormElements"
import createSubmission from "../Submission/create"
import updateSubmission from "../Submission/update"
import { uploadImageAndPassToGemini } from "../Submission/uploadPhotoToGemini"
import { LargeHeadline, Subtitle } from "../Typography"
import Artist from "./Artist"
import Confirmation from "./Confirmation"
import Edition from "./Edition"
import Location from "./Location"
import Metadata from "./Metadata"
import Provenance from "./Provenance"
import SelectFromPhotoLibrary from "./SelectFromPhotoLibrary"

const consignmentsStateKey = "ConsignmentsStoredState"

interface Props extends ViewProperties {
  navigator: NavigatorIOS
  route: Route // this gets set by NavigatorIOS
  setup: ConsignmentSetup
}

interface State extends ConsignmentSetup {
  hasLoaded?: boolean
  /** Used at the end to keep track of the final submission to convection for the Confirmation page to see */
  hasSubmittedSuccessfully?: boolean
}

const track: Track<Props, State> = _track

@screenTrack({
  context_screen: Schema.PageNames.ConsignmentsOverView,
  context_screen_owner_type: Schema.OwnerEntityTypes.Consignment,
})
export default class Overview extends React.Component<Props, State> {
  constructor(props) {
    super(props)
    this.state = props.setup || {}

    // Grab stored details from the local storage if no
    // props have been passed in
    if (!props.setup) {
      this.restoreFromLocalStorage()
    }
  }

  saveStateToLocalStorage = () => AsyncStorage.setItem(consignmentsStateKey, JSON.stringify(this.state))
  restoreFromLocalStorage = () =>
    AsyncStorage.getItem(consignmentsStateKey, (_err, result) => {
      const results = (result && JSON.parse(result)) || {}
      this.setState({ ...results, hasLoaded: true })
    })

  goToArtistTapped = () =>
    this.props.navigator.push({
      component: Artist,
      passProps: { ...this.state, updateWithArtist: this.updateArtist },
    })

  goToProvenanceTapped = () =>
    this.props.navigator.push({
      component: Provenance,
      passProps: { ...this.state, updateWithProvenance: this.updateProvenance },
    })

  goToPhotosTapped = () =>
    this.props.navigator.push({
      component: SelectFromPhotoLibrary,
      passProps: { setup: this.state, updateWithPhotos: this.updatePhotos },
    })

  goToMetadataTapped = () =>
    this.props.navigator.push({
      component: Metadata,
      passProps: { metadata: this.state.metadata, updateWithMetadata: this.updateMetadata },
    })

  goToEditionTapped = () =>
    this.props.navigator.push({
      component: Edition,
      passProps: { setup: this.state, updateWithEdition: this.updateEdition },
    })

  goToLocationTapped = () =>
    this.props.navigator.push({ component: Location, passProps: { updateWithResult: this.updateLocation } })

  updateArtist = (result: SearchResult) => this.updateStateAndMetaphysics({ artist: result })
  updateMetadata = (result: ConsignmentMetadata) => this.updateStateAndMetaphysics({ metadata: result })
  updateProvenance = (result: string) => this.updateStateAndMetaphysics({ provenance: result })
  updateEdition = (result: ConsignmentSetup) => this.updateStateAndMetaphysics(result)
  updateLocation = (city: string, state: string, country: string) =>
    this.updateStateAndMetaphysics({ location: { city, state, country } })

  updatePhotos = (photos: string[]) =>
    photos.length && this.updateStateAndMetaphysics({ photos: photos.map(f => ({ file: f, uploaded: false })) })

  updateStateAndMetaphysics = (state: Partial<ConsignmentSetup>) =>
    this.setState(state, this.updateLocalStateAndMetaphysics)

  updateLocalStateAndMetaphysics = async () => {
    this.saveStateToLocalStorage()

    if (this.state.submission_id) {
      // This is async, but we don't need the synchronous behavior from await.
      this.uploadPhotosIfNeeded()

      updateSubmission(this.state, this.state.submission_id)
    } else if (this.state.artist) {
      const submission = await createSubmission(this.state)
      this.setState({ submission_id: submission.id }, () => {
        this.submissionDraftCreated()
      })
    }
  }

  @track((_props, state) => ({
    action_type: Schema.ActionTypes.Success,
    action_name: Schema.ActionNames.ConsignmentDraftCreated,
    owner_id: state.submission_id,
    owner_type: Schema.OwnerEntityTypes.Consignment,
    owner_slug: state.submission_id,
  }))
  submissionDraftCreated() {
    return null
  }

  submitFinalSubmission = async () => {
    this.showConfirmationScreen()

    const submission = this.state as ConsignmentSetup
    let hasSubmittedSuccessfully = true
    try {
      await updateSubmission({ ...submission, state: "SUBMITTED" }, this.state.submission_id)
      await AsyncStorage.removeItem(consignmentsStateKey)
      this.submissionDraftSubmitted()
    } catch (error) {
      console.error("Overview final submission: " + error)
      hasSubmittedSuccessfully = false
    }

    this.setState({ hasSubmittedSuccessfully })
  }

  @track((_props, state) => ({
    action_type: Schema.ActionTypes.Success,
    action_name: Schema.ActionNames.ConsignmentSubmitted,
    owner_id: state.submission_id,
    owner_type: Schema.OwnerEntityTypes.Consignment,
    owner_slug: state.submission_id,
  }))
  submissionDraftSubmitted() {
    return null
  }

  showConfirmationScreen() {
    // Confirmation will ask to see how the submission process has worked in 1 second
    const submissionRequestValidationCheck = () => this.state.hasSubmittedSuccessfully
    // Show confirmation screen
    this.props.navigator.push({ component: Confirmation, passProps: { submissionRequestValidationCheck } })
  }

  exitModal = () => SwitchBoard.dismissModalViewController(this)

  uploadPhotosIfNeeded = async () => {
    const uploading = this.state.photos && this.state.photos.some(f => f.uploading)
    const toUpload = this.state.photos && this.state.photos.filter(f => !f.uploaded && f.file)
    if (!uploading && toUpload && toUpload.length) {
      // Pull out the first in the queue and upload it
      const photo = toUpload[0]

      // Set this one photo to upload, so that if you go in and out
      // quickly it doesn't upload duplicates
      photo.uploading = true
      this.setState({ photos: this.state.photos })
      await uploadImageAndPassToGemini(photo.file, "private", this.state.submission_id)

      // Mutate state 'unexpectedly', then send it back through "setState" to trigger the next
      // in the queue
      photo.uploaded = true
      photo.uploading = false
      this.setState({ photos: this.state.photos }, this.uploadPhotosIfNeeded)
    }
  }

  canSubmit = () =>
    !!(
      this.state.artist &&
      this.state.location &&
      this.state.metadata &&
      this.state.metadata.category &&
      this.state.metadata.title &&
      this.state.metadata.year &&
      this.state.metadata.medium &&
      this.state.metadata.height &&
      this.state.metadata.width &&
      this.state.editionScreenViewed
    )

  render() {
    const title = "Complete work details to submit"
    const subtitle = "Provide as much detail as possible so that our partners can best assess your work."

    // See https://github.com/artsy/convection/blob/master/app/models/submission.rb for list
    const canSubmit = this.canSubmit()

    const isPad = Dimensions.get("window").width > 700

    return (
      <ConsignmentBG>
        <ScrollView style={{ flex: 1 }} alwaysBounceVertical={false} centerContent>
          <View
            style={{
              paddingTop: 10,
              alignSelf: "center",
              width: "100%",
              maxWidth: 540,
              flex: 1,
            }}
          >
            <LargeHeadline style={{ textAlign: isPad ? "center" : "left" }}>{title}</LargeHeadline>
            <Subtitle style={{ textAlign: isPad ? "center" : "left", marginBottom: isPad ? 80 : 0, marginTop: -15 }}>
              {subtitle}
            </Subtitle>
            <View style={{ flex: 1 }}>
              <TODO
                goToArtist={this.goToArtistTapped}
                goToPhotos={this.goToPhotosTapped}
                goToEdition={this.goToEditionTapped}
                goToMetadata={this.goToMetadataTapped}
                goToLocation={this.goToLocationTapped}
                goToProvenance={this.goToProvenanceTapped}
                {...this.state}
              />
            </View>
            <Row style={{ justifyContent: "center", marginTop: isPad ? 80 : -30 }}>
              {this.state.hasLoaded && (
                <Button text="SUBMIT" onPress={canSubmit ? this.submitFinalSubmission : undefined} />
              )}
            </Row>
            <Row style={{ justifyContent: "center" }}>
              {/* TODO: This margin offset is just battling the default margins set on the buttons */}
              <CloseButton style={{ marginTop: -8 }} />
            </Row>
          </View>
        </ScrollView>
      </ConsignmentBG>
    )
  }
}
