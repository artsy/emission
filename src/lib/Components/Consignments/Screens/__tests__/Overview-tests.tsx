import React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import Artist from "../Artist"
import Edition from "../Edition"
import Location from "../Location"
import Metadata from "../Metadata"
import Overview from "../Overview"

import Provenance from "../Provenance"
import SelectFromPhotoLibrary from "../SelectFromPhotoLibrary"

const nav = {} as any
const route = {} as any

const anything = expect.anything

it("Sets up the right view hierarchy", () => {
  const tree = renderer.create(<Overview navigator={nav} route={route} setup={{}} />).toJSON()
  expect(tree).toMatchSnapshot()
})

describe("Opening the right page", () => {
  let navigator = {} as any
  let overview: Overview

  beforeEach(() => {
    navigator = { push: jest.fn() } as any
    overview = new Overview({ navigator, route, setup: {} })
  })

  it("pushes a Location when you tap on Locations", () => {
    overview.goToLocationTapped()
    expect(navigator.push).toBeCalledWith({ component: Location, passProps: anything() })
  })

  it("pushes a Artist when you tap on Artist", () => {
    overview.goToArtistTapped()
    expect(navigator.push).toBeCalledWith({ component: Artist, passProps: anything() })
  })

  it("pushes a Metadata when you tap on Metadata", () => {
    overview.goToMetadataTapped()
    expect(navigator.push).toBeCalledWith({ component: Metadata, passProps: anything() })
  })

  it("pushes a Edition when you tap on Edition", () => {
    overview.goToEditionTapped()
    expect(navigator.push).toBeCalledWith({ component: Edition, passProps: anything() })
  })

  it("pushes a Provenance when you tap on Provenance", () => {
    overview.goToProvenanceTapped()
    expect(navigator.push).toBeCalledWith({ component: Provenance, passProps: anything() })
  })

  it("pushes a SelectFromPhotoLibrary when you tap on Photos", () => {
    overview.goToPhotosTapped()
    expect(navigator.push).toBeCalledWith({ component: SelectFromPhotoLibrary, passProps: anything() })
  })
})

describe("Updating State", () => {
  const navigator: any = { push: jest.fn() }
  let overview: Overview
  let stateMock: jest.Mock<any>
  let update: any

  beforeEach(() => {
    overview = new Overview({ navigator, route, setup: {} })
    update = overview.updateLocalStateAndMetaphysics
    overview.setState = jest.fn()
    stateMock = overview.setState as any
  })

  it("updates Location", () => {
    overview.updateLocation("Huddersfield", "Yorkshire", "UK")
    expect(stateMock).toBeCalledWith({ location: { city: "Huddersfield", country: "UK", state: "Yorkshire" } }, update)
  })

  it("updates Artist", () => {
    overview.updateArtist({ id: "banksy", name: "Banksy" })
    expect(stateMock).toBeCalledWith({ artist: { id: "banksy", name: "Banksy" } }, update)
  })

  it("updates work metadata", () => {
    overview.updateMetadata({
      title: "OK",
      year: "1983",
      category: "PAINTING",
      categoryName: "Painting",
      medium: "Oil on Canvas",
      width: "100",
      height: "200",
      depth: 20,
      unit: "cm",
      displayString: "Paint on canvas",
    })

    expect(stateMock).toBeCalledWith(
      {
        metadata: {
          category: "PAINTING",
          categoryName: "Painting",
          depth: 20,
          displayString: "Paint on canvas",
          height: "200",
          medium: "Oil on Canvas",
          title: "OK",
          unit: "cm",
          width: "100",
          year: "1983",
        },
      },
      update
    )
  })

  it("updates Provenance", () => {
    overview.updateProvenance("This is my provenance")
    expect(stateMock).toBeCalledWith({ provenance: "This is my provenance" }, update)
  })
})

it("requires the same metadata props as force", () => {
  const requiredProps: any = {
    artist: {},
    location: {},
    editionScreenViewed: true,
    metadata: {
      category: "a",
      categoryName: "A",
      title: "Work",
      year: "123",
      medium: "123",
      width: "123",
      height: "12",
      depth: 12,
      unit: "CM",
      displayString: "A work",
    },
  }
  const overview = new Overview({ setup: requiredProps })
  expect(overview.canSubmit()).toBeTruthy()
})

it("does not allow submission without all the right options", () => {
  const requiredProps: any = {
    artist: {},
    location: {},
    editionScreenViewed: true,
  }
  const overview = new Overview({ setup: requiredProps })
  expect(overview.canSubmit()).toBeFalsy()
})
