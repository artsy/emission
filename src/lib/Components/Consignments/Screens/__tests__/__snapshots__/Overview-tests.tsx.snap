// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sets up the right view hierarchy 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "black",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      alwaysBounceVertical={false}
      centerContent={true}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flex": 1,
              "maxWidth": 540,
              "paddingTop": 10,
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 30,
                  "marginBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "textAlign": "center",
                },
                Object {
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }
          >
            Complete work details to submit
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#808080",
                  "fontSize": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                },
                Object {
                  "marginBottom": 80,
                  "marginTop": -15,
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }
          >
            Provide as much detail as possible so that our partners can best assess your work.
          </Text>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "black",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "paddingBottom": 20,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 20,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#444444",
                      "height": 1,
                    },
                    undefined,
                  ]
                }
              />
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={null}
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 60,
                      },
                      null,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "flexBasis": 0,
                          "flexGrow": 2,
                          "flexShrink": 1,
                          "fontFamily": "AvantGardeGothicITC",
                        },
                        undefined,
                      ]
                    }
                  >
                    ARTIST/DESIGNER NAME
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "borderColor": "white",
                          "borderRadius": 13,
                          "borderWidth": 1,
                          "height": 26,
                          "justifyContent": "center",
                          "marginRight": 20,
                          "width": 26,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../images/consignments/plus.png",
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#444444",
                      "height": 1,
                    },
                    undefined,
                  ]
                }
              />
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={null}
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 60,
                      },
                      null,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "flexBasis": 0,
                          "flexGrow": 2,
                          "flexShrink": 1,
                          "fontFamily": "AvantGardeGothicITC",
                        },
                        undefined,
                      ]
                    }
                  >
                    PHOTOS
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "borderColor": "white",
                          "borderRadius": 13,
                          "borderWidth": 1,
                          "height": 26,
                          "justifyContent": "center",
                          "marginRight": 20,
                          "width": 26,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../images/consignments/plus.png",
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#444444",
                      "height": 1,
                    },
                    undefined,
                  ]
                }
              />
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={null}
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 60,
                      },
                      null,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "flexBasis": 0,
                          "flexGrow": 2,
                          "flexShrink": 1,
                          "fontFamily": "AvantGardeGothicITC",
                        },
                        undefined,
                      ]
                    }
                  >
                    WORK DETAILS
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "borderColor": "white",
                          "borderRadius": 13,
                          "borderWidth": 1,
                          "height": 26,
                          "justifyContent": "center",
                          "marginRight": 20,
                          "width": 26,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../images/consignments/plus.png",
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#444444",
                      "height": 1,
                    },
                    undefined,
                  ]
                }
              />
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={null}
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 60,
                      },
                      null,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "flexBasis": 0,
                          "flexGrow": 2,
                          "flexShrink": 1,
                          "fontFamily": "AvantGardeGothicITC",
                        },
                        undefined,
                      ]
                    }
                  >
                    EDITION & AUTHENTICITY
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "borderColor": "white",
                          "borderRadius": 13,
                          "borderWidth": 1,
                          "height": 26,
                          "justifyContent": "center",
                          "marginRight": 20,
                          "width": 26,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../images/consignments/plus.png",
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#444444",
                      "height": 1,
                    },
                    undefined,
                  ]
                }
              />
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={null}
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 60,
                      },
                      null,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "flexBasis": 0,
                          "flexGrow": 2,
                          "flexShrink": 1,
                          "fontFamily": "AvantGardeGothicITC",
                        },
                        undefined,
                      ]
                    }
                  >
                    LOCATION
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "borderColor": "white",
                          "borderRadius": 13,
                          "borderWidth": 1,
                          "height": 26,
                          "justifyContent": "center",
                          "marginRight": 20,
                          "width": 26,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../images/consignments/plus.png",
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#444444",
                      "height": 1,
                    },
                    undefined,
                  ]
                }
              />
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={null}
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 60,
                      },
                      null,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "flexBasis": 0,
                          "flexGrow": 2,
                          "flexShrink": 1,
                          "fontFamily": "AvantGardeGothicITC",
                        },
                        undefined,
                      ]
                    }
                  >
                    PROVENANCE
                  </Text>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "borderColor": "white",
                          "borderRadius": 13,
                          "borderWidth": 1,
                          "height": 26,
                          "justifyContent": "center",
                          "marginRight": 20,
                          "width": 26,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../images/consignments/plus.png",
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#444444",
                      "height": 1,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "justifyContent": "center",
                  "marginTop": 80,
                },
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingVertical": 6,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingVertical": 6,
                },
              ]
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(0, 0, 0, 1)",
                  "borderColor": "rgba(0, 0, 0, 1)",
                  "borderWidth": 1,
                  "height": 43,
                  "justifyContent": "center",
                  "marginTop": -8,
                  "width": 174,
                }
              }
              testID={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontSize": 12,
                    },
                    Object {
                      "color": "rgba(255, 255, 255, 1)",
                      "fontFamily": "AGaramondPro-Regular",
                      "opacity": 1,
                    },
                    Object {
                      "fontFamily": "AvantGardeGothicITC",
                    },
                  ]
                }
              >
                CLOSE
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
